packages/foam-vscode/src/services/templates.ts

```diff
@@ -246,15 +334,32 @@ export const NoteFactory = {
    * Creates a new note when following a placeholder wikilink using the default template.
    * @param wikilinkPlaceholder the placeholder value from the wikilink. (eg. `[[Hello Joe]]` -> `Hello Joe`)
    * @param filepathFallbackURI the URI to use if the template does not specify the `filepath` metadata attribute. This is configurable by the caller for backwards compatibility purposes.
+   * @param chooseTemplate whether the user should be prompted to choose a template for the newly created note
    */
-  createForPlaceholderWikilink: (
+  createForPlaceholderWikilink: async (
     wikilinkPlaceholder: string,
-    filepathFallbackURI: URI
+    filepathFallbackURI: URI,
+    chooseTemplate: boolean
```

Comment:
I was thinking here we might use the same pattern we have for `onFileExists`, and have an `async getTemplate` function that the client can pass in. The default implementation just returns the default template, but we can pass in your `askUserForTemplate` implementation when needed. wdyt?

Reply:
(or maybe this can be handled at the `CREATE_NOTE` level, I guess we'd have to see it to get a good sense of it)

Reply:
So the template can either be a template file (in which case we use it), a function (in which case we run the function to get a template), or nothing (then we choose the default template)?

I guess it'd work, but maybe this would be easier / make more sense at the `CREATE_NOTE` level like you say. Here we could just accept a template or choose the default one if it's not provided. Then `CREATE_NOTE` could have a way of asking the user for a template if it isn't provided?

Reply:
That sounds about right. I am wondering whether there will be a problem with the `asUri` function, so maybe we need a string (like "ask") that is interpreted in the command as prompting the user.
The default behavior should be to use the default template, to ask the user the proper value needs to be provided.
We might need to change `templatePath` as it's becoming more than just the path. Maybe we can have a `template` property, which:
- if `ask` asks for the template
- otherwise will interpret the name as the path to the template

it should work?

Reply:
Maybe would be weird if one wanted to create a note called `ask`? But otherwise should work the same

Reply:
I guess you mean if you wanted a template called "ask"? I am ok with the tradeoff

For the `template` I am really thinking of having this priority list (out of scope for this PR, but helps you understand the thinking):

- not present: don't use template
- 'ask': ask the user
- <template-name>: if it matches the name of a template (from its metadata), use that
- <template-path>: if it matches the path of a file/template, use that
- otherwise don't use template (equivalent to not present)

I understand the risk of overlapping in the namespace, but the overlap amongst the options is quite small and "workaroundable" IMO, so I am ok with that.

Reply:
After looking at the code, I like your approach. 

I might update it in the future, with a proper deprecation/migration path (which I would need to do anyways for the `templatePath` prop), but it looks pretty solid now.